port/cpl_recode_iconv.cpp

```diff
@@ -236,7 +236,7 @@ char *CPLRecodeFromWCharIconv( const wchar_t *pwszSource,
 
     sConv = iconv_open( pszDstEncoding, pszSrcEncoding );
 
-    if( sConv == reinterpret_cast<iconv_t>(-1) )
+    if( sConv == (iconv_t)(-1) )
```

Comment:
```suggestion
```suggestion
#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wold-style-cast"
#endif
    // iconv_t might be a integer or a pointer, so we have to fallback to C-style cast
    if( sConv == (iconv_t)(-1) )
#ifdef __GNUC__
#pragma GCC diagnostic pop
#endif
```
```